src/shims/posix/fs.rs

```diff
@@ -489,22 +528,32 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             // thus they can share the same implementation here.
             let &[_, _, start] = check_arg_count(args)?;
             let start = this.read_scalar(start)?.to_i32()?;
-            if fd < MIN_NORMAL_FILE_FD {
-                throw_unsup_format!("duplicating file descriptors for stdin, stdout, or stderr is not supported")
-            }
+
             let fh = &mut this.machine.file_handler;
-            let (file_result, writable) = match fh.handles.get(&fd) {
+
+            match fh.handles.get(&fd) {
                 Some(file_descriptor) => {
-                    // FIXME: Support "dup" for all FDs(stdin, etc)
-                    let FileHandle { file, writable } = file_descriptor.as_file_handle()?;
-                    (file.try_clone(), *writable)
+                    let dup_result : io::Result<Box<dyn FileDescriptor>> =
+                        if let Ok(FileHandle { file, writable }) = file_descriptor.as_file_handle() {
+                            file.try_clone()
+                                .map(|dup_file|
+                                     Box::new(FileHandle { file: dup_file,
+                                                           writable: *writable }) as Box<dyn FileDescriptor>)
+                        } else {
+                            match fd {
+                                0 => Ok(Box::new(io::stdin())),
+                                1 => Ok(Box::new(io::stdout())),
+                                2 => Ok(Box::new(io::stderr())),
+                                _ => return this.handle_not_found(),
```

Comment:
There should be no `match` on the file descriptor anywhere. That is the entire point of this change!